code/modules/antagonists/spy/spy_uplink.dm

```diff
@@ -56,7 +56,7 @@
 /datum/component/spy_uplink/proc/on_attack_self(obj/item/source, mob/user)
 	SIGNAL_HANDLER
 
-	if(is_our_spy(user))
+	if(user.mind.has_antag_datum(/datum/antagonist/spy))
```

Comment:
Can you add an `IS_SPY()` helper like the other antags and use it here

Reply:
We generally don't do this for other antag datum checks, are you sure?